### PR TITLE
Ui

### PR DESCRIPTION
**Pull request by naota-nandaba**
- write post component
- display posts component
- nav bar component
- footer component

### CODE REVIEW

**Comment by naota-nandaba** [comment]
@OfTheDelmer

File: client/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    return request({
+      method: method,
+      qs: resource.query,
+      uri: this.uri.concat(resource.path),
+      headers: resource.skipAuth,
+      body: body,
+      json: true
+     })
+  }
+
```
Comment (OfTheDelmer) [inline client/index.js:31]: Add ESLint for AirBnB style.

File: client/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "client",
+  "version": "1.0.0",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (OfTheDelmer) [inline client/package.json:6]: Remove broken `test` script.

File: ui/package.json [hunk 3]
```diff
@@ -25,7 +36,8 @@
     "babel-loader": "^7.1.2",
     "babel-preset-es2015": "^6.24.1",
     "babel-preset-react": "^6.24.1",
-    "react-dom": "^15.6.1",
+    "git-open": "^1.3.1",
+    "url-loader": "^0.5.9",
```
Comment (OfTheDelmer) [inline ui/package.json:40]: The other loaders should also be moved to the devDependencies.

File: ui/public/css/app.css [hunk 4]
```diff
@@ -1,44 +1,154 @@
...
   text-align: right;
+  align-items: right;
+  align-self: right;
+  justify-content: right;
+  font-family: Helvetica;
+  font-size: 14px;
 }
 
 /*DreamFeed*/
+
```
Comment (OfTheDelmer) [inline ui/public/css/app.css:112]: These should moved to a separate `dream_feed.css`, `dream_form.css`, `dream.css`, etc.

File: ui/public/css/app.css [hunk 5]
```diff
@@ -1,44 +1,154 @@
+@media (min-width: 500px) {
+    .container{
+        max-width: 500px;
+    }
+}
+
+@import url('https://fonts.googleapis.com/css?family=Raleway');
```
Comment (OfTheDelmer) [inline ui/public/css/app.css:7]: Should just do this using one `@import`:

```css
@import url('https://fonts.googleapis.com/css?family=Lato|Raleway');
```

File: ui/public/css/app.css [hunk 6]
```diff
@@ -1,44 +1,154 @@
...
+  font-family: Helvetica;
+  font-size: 14px;
 }
 
 p.char_count{
   text-align: right;
+  align-items: right;
+  align-self: right;
+  justify-content: right;
+  font-family: Helvetica;
```
Comment (OfTheDelmer) [inline ui/public/css/app.css:107]: Is there any reason this isn't `Lato` or `Raleway`? Violating repetition of fonts just for form elements seems a bit odd.

File: ui/public/css/app.css [hunk 7]
```diff
@@ -1,44 +1,154 @@
...
+        max-width: 500px;
+    }
+}
+
+@import url('https://fonts.googleapis.com/css?family=Raleway');
+@import url('https://fonts.googleapis.com/css?family=Lato');
+
+body{
+  background-color: #487190;
+  font-family: 'Helvetica';
```
Comment (OfTheDelmer) [inline ui/public/css/app.css:12]: You don't seem to be using `Helvetica` nor `Lato` here.

File: ui/public/js/a_dream.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import ReactDOM from 'react-dom';
+import { Router, IndexRoute, Route, hashHistory } from 'react-router'
+const storiesOf = require('@storybook/react').storiesOf;
```
Comment (OfTheDelmer) [inline ui/public/js/a_dream.js:4]: `import { storiesOf }  from '@stroybook/react';`

File: ui/public/js/add_dream.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+                     />
+
+              <button type="button"
+                      className="btn btn-primary add_btn"
+                      disabled={this.state.disabled}>
+                      Post</button>
+
+              <text
+                className="char_count"
+                style={this.state.char_color}>
```
Comment (OfTheDelmer) [inline ui/public/js/add_dream.js:42]: Camel-casing here.

File: ui/public/js/add_dream.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+                {this.props.charCount - this.state.post.body.length}</text>
```
Comment (OfTheDelmer) [inline ui/public/js/add_dream.js:43]: Could benefit from being moved into a method `DreamForm#remainingCount`

```js
remainingCount() {
   return this.props.charCount - this.state.post.body.length;
}
```

File: ui/public/js/app.js [hunk 11]
```diff
@@ -1,21 +1,27 @@
...
+import 'bootstrap/dist/css/bootstrap.css';
+import '../css/app.css'
 
 const App = React.createClass({
   render: function () {
     return (
-      <h1>Hello world</h1>
+      <div>
+        <div className="container nav_contain">
+        <NavBar />
```
Comment (OfTheDelmer) [inline ui/public/js/app.js:15]: Indentation required.

File: ui/public/js/dream_feed.js [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+    if (query.refresh) {
+      this.refreshPosts();
+      console.log('Refreshing');
+    }
+
+    const dreamList = this.state.posts.map((val) =>(
+      <div>
+      <Dream value={val} />
+        <p></p>
```
Comment (OfTheDelmer) [inline ui/public/js/dream_feed.js:36]: Empty `<p></p>`.

File: ui/public/js/footer.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import ReactDOM from 'react-dom';
+import { Router, IndexRoute, Route, hashHistory } from 'react-router'
+const storiesOf = require('@storybook/react').storiesOf;
+
+import 'bootstrap/dist/css/bootstrap.css';
+import '../css/app.css'
+
+const Footer = React.createClass({
```
Comment (OfTheDelmer) [inline ui/public/js/footer.js:9]: I like that you added a footer component.

Comment (OfTheDelmer) [reply ui/public/js/footer.js:9]: Could probably just be a stateless component:

```js
const Footer = () => (
   // render code
);
```

File: ui/public/js/header.js [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+const NavBar = React.createClass({
+  render: function () {
+    return (
+
+        <nav className="navbar">
+
+            <div className="navbar-header">
+
+            <ul className="nav nav-pills" role="tablist">
+              <li role="presentation" className="active"> <Link to="/#">My Mu</Link></li>
```
Comment (OfTheDelmer) [inline ui/public/js/header.js:19]: [Take a look at React Router's active styling](https://github.com/reactjs/react-router-tutorial/tree/master/lessons/05-active-links)

File: ui/stories/a_dream.js [hunk 15]
```diff
@@ -20,7 +19,7 @@ const Dream = React.createClass({
 storiesOf('a Dream', module)
 .add('(Stateless) Default', () => (
   <div className="container">
-    <div className="col-xs-6 jumbotron row dream_feed">
+    <div className="col-xs-12 jumbotron row dream_feed">
```
Comment (OfTheDelmer) [inline ui/stories/a_dream.js:22]: Nice to see Bootstrap grids for your layout.

Comment (OfTheDelmer) [reply ui/stories/a_dream.js:22]: I like that you're also trying to keep your stories up to date.

**Review by OfTheDelmer — COMMENTED** [review]
It looks like it's coming along for you. I would create issues and add tasks to your project tracker to resolve some of the comments I left. Adding ESLint will help with some of it down the road.

**Pull request merged**